lib/spack/spack/build_environment.py

```diff
@@ -133,6 +133,24 @@ def should_set_parallel_jobs(jobserver_support=False):
     return not env_flag(SPACK_NO_PARALLEL_MAKE)
 
 
+class ModuleChangeAccumulator(object):
+    def __init__(self):
+        self.changes = {}
+
+    def __setattr__(self, key, value):
+        if key == "changes":
+            super(ModuleChangeAccumulator, self).__setattr__(key, value)
+        else:
+            self.changes[key] = value
+
+    def __getattr__(self, key):
+        # Return what's in our dict, else delegate to superclass
+        if key in self.changes:
+            return self.changes[key]
+        else:
+            return super(ModuleChangeAccumulator, self).__getattribute__(key)
```

Comment:
I think you can replace this with a `spack.util.pattern.Bunch` and use `__dict__` to get the attributes.

Reply:
Great idea, I had no idea we had that utility. Cleans this up a ton